src/angular-app/languageforge/lexicon/js/lexicon.services.rights.js

```diff
@@ -1,67 +1,59 @@
-        return false;
-
-      return (sessionService.session.project.userIsProjectOwner ||
-        sessionService.hasSiteRight(sessionService.domain.PROJECTS,
-          sessionService.operation.ARCHIVE));
-    };
-
-    this.canEditProject = function canEditProject() {
-      if (sendReceive.isInProgress() || sessionService.session.project.isArchived) return false;
-
-      return sessionService.hasProjectRight(sessionService.domain.PROJECTS,
-        sessionService.operation.EDIT);
-    };
-
-    this.canEditEntry = function canEditEntry() {
-      if (sendReceive.isInProgress() || sessionService.session.project.isArchived) return false;
-
-      return sessionService.hasProjectRight(sessionService.domain.ENTRIES,
-        sessionService.operation.EDIT);
-    };
-
-    this.canDeleteEntry = function canDeleteEntry() {
-      if (sendReceive.isInProgress() || sessionService.session.project.isArchived) return false;
-
-      return sessionService.hasProjectRight(sessionService.domain.ENTRIES,
-        sessionService.operation.DELETE);
-    };
-
-    this.canComment = function canComment() {
-      if (sendReceive.isInProgress() || sessionService.session.project.isArchived) return false;
-
-      return sessionService.hasProjectRight(sessionService.domain.COMMENTS,
-        sessionService.operation.CREATE);
-    };
-
-  }])
-
-  ;
+    var session;
+
+    /**
+     * This helper returns a function with a boolean return type indicating whether an
+     * operation is permitted.
+     * @param {boolean} allowArchived - When false, the resulting function will always return false
+     * if the project is archived.
+     * @param {number} domain - An enum value from sessionService.domain indicating the domain of
+     * the operation.
+     * @param {number} operation - An enum value from sessionService.operation indicating the
+     * operation for which the user's permissions should be checked.
+     * @param {projectOwnerAllowed} - When this is true and the current project is owned by the
+     * current user, the user will be considered to always have permission.
+     * @return {Function<boolean>} - A function that will indicate whether the user is
+     * allowed to perform the given operation.
+     */
+    function condition(allowArchived, domain, operation, projectOwnerAllowed) {
+      return function() {
+        if(sendReceive.isInProgress()) return false;
+        else if(!session.project()) return false;
+        else if(!allowArchived && session.project().isArchived) return false;
+        else {
+          var hasRight = session.hasProjectRight(domain, operation);
+          // The case where user does not explicitly have a right, but does because user is owner
+          if(projectOwnerAllowed) hasRight = hasRight || session.project().userIsProjectOwner;
+          return hasRight;
+        }
+      };
+    }
+
+    var Rights = (new function () {
```

Comment:
Indeed. I think the session service has grown into "stuff that's convenient to have globally" rather than strictly session data. Now that it does all this it could use some refactoring into multiple services.